inc/saihostintf.h

```diff
@@ -461,120 +469,23 @@ typedef sai_status_t(*sai_get_hostif_trap_attribute_fn)(
-
-    /** fdb traps */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_FDB_MIN = SAI_HOSTIF_USER_DEFINED_TRAP_ID_NEIGH_MAX + 1,
-
-    /** fdb traps max */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_FDB_MAX = SAI_HOSTIF_USER_DEFINED_TRAP_ID_FDB_MIN + SAI_HOSTIF_USER_DEFINED_TRAP_ID_TABLE_RANGE,
-
-    /** Custom range base */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ID_CUSTOM_RANGE_BASE = 0x10000000,
-
-} sai_hostif_user_defined_trap_id_t;
-
-/**
- * @brief Host interface user defined traps attributes
+ * @brief Attribute data for SAI_HOSTIF_ATTR_TYPE
  */
-typedef enum _sai_hostif_user_defined_trap_attr_t
+typedef enum _sai_hostif_type_t
 {
-    /**
-     * @brief Start of attributes
-     */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ATTR_START,
+    /** Port-based Host Interface Type */
+    SAI_HOSTIF_TYPE_PORT,
 
-    /**
-     * @brief Trap channel to use
-     *
-     * @type sai_hostif_trap_channel_t
-     * @flags CREATE_ONLY
-     * @default SAI_HOSTIF_TRAP_CHANNEL_CB
-     */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ATTR_HOSTIF_TRAP_CHANNEL = SAI_HOSTIF_USER_DEFINED_TRAP_ATTR_START,
+    /** LAG based Host Interface Type */
+    SAI_HOSTIF_TYPE_LAG,
 
-    /**
-     * @brief File descriptor
-     *
-     * Valid only when #SAI_HOSTIF_TRAP_ATTR_TRAP_CHANNEL == #SAI_HOSTIF_TRAP_CHANNEL_FD
-     * Must be set before set #SAI_HOSTIF_TRAP_ATTR_TRAP_CHANNEL to #SAI_HOSTIF_TRAP_CHANNEL_FD
-     *
-     * @type sai_object_id_t
-     * @objects SAI_OBJECT_TYPE_HOSTIF
-     * @flags CREATE_AND_SET
-     */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ATTR_FD,
-
-    /**
-     * @brief End of attributes
-     */
-    SAI_HOSTIF_USER_DEFINED_TRAP_ATTR_END,
-
-} sai_hostif_user_defined_trap_attr_t;
-
-/**
- * @brief Set user defined trap attribute value.
- *
- * @param[in] hostif_user_defined_trap_id Host interface user defined trap id
- * @param[in] attr Aattribute
- *
- * @return #SAI_STATUS_SUCCESS on success Failure status code on error
- */
-typedef sai_status_t(*sai_set_hostif_user_defined_trap_attribute_fn)(
-        _In_ sai_hostif_user_defined_trap_id_t hostif_user_defined_trapid,
-        _In_ const sai_attribute_t *attr);
-
-/**
- * @brief Get user defined trap attribute value.
- *
- * @param[in] hostif_user_defined_trap_id Host interface user defined trap id
- * @param[in] attr_count Number of attributes
- * @param[inout] attr_list Array of attributes
- *
- * @return #SAI_STATUS_SUCCESS on success Failure status code on error
- */
-typedef sai_status_t(*sai_get_hostif_user_defined_trap_attribute_fn)(
-        _In_ sai_hostif_user_defined_trap_id_t hostif_user_defined_trapid,
-        _In_ uint32_t attr_count,
-        _Inout_ sai_attribute_t *attr_list);
+    /** Vlan Router Interface based Host Interface Type */
+    SAI_HOSTIF_TYPE_VLAN,
 
-/**
- * @brief Attribute data for SAI_HOSTIF_ATTR_TYPE
- */
-typedef enum _sai_hostif_type_t
-{
-    /** Netdev-based Host Interface Type */
+    /** Netdevice */
     SAI_HOSTIF_TYPE_NETDEV,
```

Comment:
@tushar-ty , SAI_HOSTIF_TYPE_PORT/LAG/VLAN should be removed according to the latest doc.